lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -507,18 +548,31 @@ export default class BitcoinProcessor {
 
     // console.debug(`Block ${block} contains ${transactions.length} transactions`);
 
+    // reseed source of psuedo-randomness to the blockhash
+    this.transactionSampler.resetPsuedoRandomSeed(blockHash);
+
     // iterate through transactions
     for (let transactionIndex = 0; transactionIndex < transactions.length; transactionIndex++) {
+      const transaction = transactions[transactionIndex];
+
       // get the output coins in the transaction
-      if (!('vout' in transactions[transactionIndex])) {
+      if (!('vout' in transaction)) {
         // console.debug(`Skipping transaction ${transactionIndex}: no output coins.`);
         continue;
       }
 
-      try {
-        const outputs = transactions[transactionIndex].vout as Array<any>;
+      // Add the transaction to the sampler.  We filter out transactions with unusual
```

Comment:
As per our offline discussion, please filter out the 'sidetree' transactions as well.

Reply:
Done.